service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -152,6 +154,25 @@ public void operationComplete(ConnectFuture future) {
             }
         }
 
+        /**
+         * Helper method performing loop detection
+         * @param connectSession
+         * @param viaHeaders
+         */
+        private void performLoopDetection(DefaultHttpSession connectSession, List<String> viaHeaders) {
```

Comment:
I think this can be done on the accept session itself (just after validateRequestPath) that way no need to connect to the next server. You could also simplify the logic
```
if (viaHeaders != null && viaHeaders.stream().anyMatch(h -> h.equals(viaHeader))) {
...
```